conformance/test/test_server.py

```diff
@@ -0,0 +1,60 @@
+import subprocess
+import sys
+from pathlib import Path
+
+import pytest
+from _util import maybe_patch_args_with_debug
+
+_current_dir = Path(__file__).parent
+_server_py_path = str(_current_dir / "server.py")
+_config_path = str(_current_dir / "config.yaml")
+
+
+def test_server_sync():
+    args = maybe_patch_args_with_debug(
+        [sys.executable, _server_py_path, "--mode", "sync"]
+    )
+    result = subprocess.run(
+        [
+            "go",
+            "tool",
+            "connectconformance",
+            "--conf",
+            _config_path,
+            "--mode",
+            "server",
+            "--parallel",
+            "1",
```

Comment:
I couldn't get this to pass on MacOS default `ulimit` of 256 files without setting to just 1. I debugged hypercorn shutdown a lot since cancelling asyncio properly can be tricky, but anyways it's a subprocess so it seemed weird.

In the end, I think the issue is on the conformance test side, here

https://github.com/connectrpc/conformance/blob/main/internal/app/referenceclient/client.go#L183